### PR TITLE
Fix Circle merge on master branch.

### PR DESCRIPTION
**Pull request by QuLogic**
## PR Summary

There exists a PR on a random fork, which fills in `CI_PULL_REQUEST`, which we don't want to use as merge target when building `master`. Instead, check `CIRCLE_PR_NUMBER`, which is set only on forked PRs. Also, check branch name (which is current, not the _target_) to be certain.

Followup to #20680, @larsoner 

## PR Checklist

- [n/a] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [n/a] New features are documented, with examples if plot related.
- [n/a] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [n/a] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [n/a] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [n/a] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

**Comment by tacaswell** [comment]
We should wait for circle to finish and then merge.  We won't see if this actually fixed the problem until it lands on master.

**Review by tacaswell — APPROVED** [review]
anyone can merge when CircleCI is green.

**Comment by larsoner** [comment]
> There exists a PR on a random fork, which fills in CI_PULL_REQUEST, which we don't want to use as merge target when building master

Wow, I've never seen that crop up before! Your solution sounds reasonable to me

**Comment by timhoffm** [comment]
Let's try this.

**Comment by jklymak** [comment]
ooops

**Comment by timhoffm** [comment]
GitHub is not thread-safe. 🤣

**Comment by QuLogic** [comment]
It was successful, I believe: [this PR merged](https://app.circleci.com/pipelines/github/matplotlib/matplotlib/10608/workflows/05272a72-6489-4be3-bd71-115ea0f9b5a0/jobs/62485) while [master didn't](https://app.circleci.com/pipelines/github/matplotlib/matplotlib/10609/workflows/531e90a7-a4f0-4750-86db-d7ba88a48398/jobs/62486).

**Pull request merged**
